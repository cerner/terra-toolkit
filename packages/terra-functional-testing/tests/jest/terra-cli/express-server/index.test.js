jest.mock('../../../../src/terra-cli/express-server/expressServerHandler');

const yargs = require('yargs');
const expressHandler = require('../../../../src/terra-cli/express-server/expressServerHandler');
const ExpressServerCommand = require('../../../../src/terra-cli/express-server');

describe('express-server', () => {
  it('declares the express-server terra-cli command with proper top level help', async () => {
    const parser = yargs.command(ExpressServerCommand).scriptName('terra').help();
    const helpOutput = await new Promise((resolve) => {
      parser.parse('--help', (err, argv, output) => {
        resolve(output);
      });
    });
    expect(helpOutput).toMatchSnapshot();
  });

  it('declares the express-server terra-cli command with proper command level help', async () => {
    const parser = yargs.command(ExpressServerCommand).scriptName('terra').help();
    const helpOutput = await new Promise((resolve) => {
      parser.parse('express-server --help', (err, argv, output) => {
        resolve(output);
      });
    });
    expect(helpOutput).toMatchSnapshot();
  });

  it('uses the appropriate handler when calling express-server', async () => {
    expressHandler.mockResolvedValueOnce();
    await ExpressServerCommand.handler({ site: 'site', host: 'host', port: 8000 });
    expect(expressHandler).toHaveBeenCalledWith(expect.objectContaining({ site: 'site', host: 'host', port: 8000 }));
  });
});
