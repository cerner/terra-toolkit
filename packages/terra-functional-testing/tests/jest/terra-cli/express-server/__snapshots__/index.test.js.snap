// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`express-server declares the express-server terra-cli command with proper command level help 1`] = `
"terra express-server

Start a simple express-server hosting a directory

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
  --port     The port on which to host the server       [number] [default: 8080]
  --host     The host on which to host the server    [string] [default: 0.0.0.0]
  --site     The directory path from which to serve the static assets
                                                             [string] [required]"
`;

exports[`express-server declares the express-server terra-cli command with proper top level help 1`] = `
"terra [command]

Commands:
  terra express-server  Start a simple express-server hosting a directory

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]"
`;
